internal/resources/slo/resource_slo.go

```diff
@@ -0,0 +1,412 @@
+							Required:     true,
+						},
+						"freeformquery": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"labels": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: `Additional labels that will be attached to all metrics generated from the query. These labels are useful for grouping SLOs in dashboard views that you create by hand.`,
+				Elem:        keyvalueSchema,
+			},
+			"objectives": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: `Over each rolling time window, the remaining error budget will be calculated, and separate alerts can be generated for each time window based on the SLO burn rate or remaining error budget.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"value": &schema.Schema{
+							Type:         schema.TypeFloat,
+							Required:     true,
+							ValidateFunc: validation.FloatBetween(0, 1),
+							Description:  `Value between 0 and 1. If the value of the query is above the objective, the SLO is met.`,
+						},
+						"window": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  `A Prometheus-parsable time duration string like 24h, 60m. This is the time window the objective is measured over.`,
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`^\d+(ms|s|m|h|d|w|y)$`), "Objective window must be a Prometheus-parsable time duration"),
+						},
+					},
+				},
+			},
+			"dashboard_uid": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `A reference to a dashboard that the plugin has installed in Grafana based on this SLO. This field is read-only, it is generated by the Grafana SLO Plugin.`,
+			},
+			"alerting": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Description: `Configures the alerting rules that will be generated for each
+				time window associated with the SLO. Grafana SLOs can generate
+				alerts when the short-term error budget burn is very high, the
+				long-term error budget burn rate is high, or when the remaining
+				error budget is below a certain threshold.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"labels": &schema.Schema{
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: `Labels will be attached to all alerts generated by any of these rules.`,
+							Elem:        keyvalueSchema,
+						},
+						"annotations": &schema.Schema{
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: `Annotations will be attached to all alerts generated by any of these rules.`,
+							Elem:        keyvalueSchema,
+						},
+						"fastburn": &schema.Schema{
+							Type:        schema.TypeList,
```

Comment:
Ordering doesn't matter for annotations or labels, so I think these should be TypeSet